src/Chapter3.hs

```diff
@@ -374,6 +381,22 @@ after the fight. The battle has the following possible outcomes:
 
 -}
 
+data Knight' = Knight' { knightHealth' :: Int
+                       , knightAttack' :: Int
+                       , knightGold' :: Int
+                       } deriving Show
+
+data Monster' = Monster' { monsterHealth' :: Int
+                         , monsterAttack' :: Int
+                         , monsterGold' :: Int
+                         } deriving Show
+
+fight' :: Monster' -> Knight' -> Int
+fight' Monster' {..} Knight' {..}
+  | knightAttack' == monsterHealth' = knightGold' + monsterGold'
+  | monsterAttack' == knightHealth' = -1
+  | otherwise                       = knightGold'
```

Comment:
Nice usage of `RecordWildCards`! Neat code :gem: 

Shameless plug, if you want to learn more about best practices around `RecordWildCards`, I can suggest my blog post on the subject:

* https://kodimensional.dev/recordwildcards

Reply:
Thank you for the link I had absolutely not idea that you could use it as a constructor, that's really cool. Also, the strict field part was very helpful in general. Seems like you should always pair `RecordWildCards` with ` StrictData` and then just selectively declare fields  as lazy, if needed.

I think those two extensions make for a great way to handle records! Despite what I've heard about records in Haskell.

I added `StrictData` as an improvement.

Reply:
Yes, `StrictData` and `RecorldWildCards` are a powerful duo :muscle: It makes sense to enable `StrictData` by default in general since it helps to avoid space-leaks. Though, it's not a standard convention yet, and we usually write a strict annotation on each field.

`RecordWildCards` are very convenient, and you need to use special naming conventions to not be confused. But so far, `RecordWildCards` provide the most convenient way to construct values, AFAIK.